code/modules/projectiles/gun_system.dm

```diff
@@ -817,8 +817,16 @@ and you're good to go.
 			sentry_battery.forceMove(get_turf(src))
 			sentry_battery.charge = 0
 			sentry_battery = null
+	// Akimbo code
+	var/obj/item/weapon/gun/active_gun = gun_user.get_active_held_item() // Don't use active or inactive_gun outside of dual_wield since there are no checks for if it's not a gun else where
+	var/obj/item/weapon/gun/inactive_gun = gun_user.get_inactive_held_item()
+	if(!dual_wield || dual_wield && gun_user.shoot_inactive_hand && active_gun.current_mag?.current_rounds > 0)
+		gun_user.shoot_inactive_hand = FALSE
```

Comment:
setting back shoot_inactive hand should be done when dual wielding is set.

remove the !dual_wield here, reset shoot_inactive_hand at line 628

Reply:
Also, very important, you must apply a delay on the next gun that will shoot. Else this is just normal akimbo but slightly slowler

Reply:
Shoot inactive hand reset on line 628